pkg/util/camel/camel_runtime_catalog.go

```diff
@@ -232,3 +230,38 @@ func (c *RuntimeCatalog) IsResolvable(uri string) bool {
 
 	return true
 }
+
+func jolokiaCatalogWorkaround() v1.Capability {
```

Comment:
Is this workaround because the jolokia-agent-jvm is resolved from a GAV classifier and not supported in the MavenArtifact struct ? 
I ask this because, a couple of weeks ago I investigated a jolokia issue and tried jolokia 2.0 in camel-k, but later on I verified camel-quarkus actually uses jolokia 1.7.2, then I parked my changes in Camel K. 
That investigation led me to add support for a GAV classifier in the MavenArtifact struct and the camel-k-catalog in camel-k-runtime, with the end goal of supporting the GAV classifier.
If interest I can share this branch.

Reply:
It's a workaround. I expect the new catalog to manage those dependencies and the possibility to include the classifier as well. I've already collected as a TODO for when the draft is completed.

Reply:
If you want to have a look at the [GAV classifier change](https://github.com/apache/camel-k/commit/f48806818e9b68d6d096030871285c99efcbbcc3) and in [camel-k-runtime](https://github.com/apache/camel-k-runtime/commit/63aed1030d99075ddae06684445903da0ea0707d), but I'm afraid this cannot yet be pushed as camel-quarkus is still in jolokia 1.7.2.